service/frontend/fx.go

```diff
@@ -267,6 +269,7 @@ func GrpcServerOptionsProvider(
 		namespaceLogInterceptor.Intercept, // TODO: Deprecate this with a outer custom interceptor
 		metrics.NewServerMetricsContextInjectorInterceptor(),
 		authInterceptor.Intercept,
+		namespaceHandoverInterceptor.Intercept,
 		redirectionInterceptor.Intercept,
```

Comment:
I guess theoretically we can retry in redirection Interceptor? since it has the knowledge of whether redirect will actually happen or not?

Reply:
I guess so but the logic could be weird. What should we do if the redirect will not happen?